plum/method.py

```diff
@@ -77,13 +82,120 @@ def __hash__(self):
         function_name = self.function_name
         signature = self.signature
         return_type = self.return_type
         impl = self.implementation
         return f"Method({function_name=}, {signature=}, {return_type=}, {impl=})"
 
+    def __rich_console__(self, console, options):
+        yield self.repr_mismatch()
+
+    def repr_mismatch(
+        self,
+        mismatches: Set[int] = frozenset(),
+        varargs_matched: bool = True,
+    ) -> str:
+        """Version of `__repr__` that can print which arguments are mismatched. This
+        is mainly used in hints.
+
+        Args:
+            mismatches (set[int], optional): Indices of the positional arguments which
+                are mismatched. Defaults to no mismatched arguments.
+            varargs_matched (bool, optional): Whether the varargs are matched. Defaults
+                to `True`.
+
+        Returns:
+            list:
+                :mod:`rich` representation of the method showing which arguments
+                are mismatched.
+        """
+        sig = self.signature
+        arg_names, kw_names, kw_var_name = extract_arg_names(self.implementation)
+
+        parts = []
+
+        # Walk through the positional arguments.
+        if sig.types:
+            for i, (arg_name, t) in enumerate(zip(arg_names, sig.types)):
+                arg_txt = Text(f"{arg_name}: ")
+                type_txt = repr_type(t)
+                if i in mismatches:
+                    type_txt.stylize("red")
+                arg_txt.append(type_txt)
+                parts.append(arg_txt)
+
+        # Print the varargs.
+        if sig.varargs != Signature._default_varargs:
+            arg_txt = Text(f"*{arg_names[-1]}: ")
+            type_txt = repr_type(sig.varargs)
+            if not varargs_matched:
+                type_txt.stylize("red")
+            arg_txt.append(type_txt)
+            parts.append(arg_txt)
+
+        if kw_names or kw_var_name is not None:
+            parts.append(Text("*"))
+        for kw_name in kw_names:
+            parts.append(Text(f"{kw_name}"))
+        if kw_var_name is not None:
+            parts.append(Text(f"**{kw_var_name}"))
+
+        res = Text(self.function_name) + Text("(") + Text(", ").join(parts) + Text(")")
+        if self.return_type != Method._default_return_type:
+            res.append(" -> ")
+            res.append(repr_type(self.return_type))
+        if sig.precedence != Signature._default_precedence:
+            res.append(f"\n    precedence={sig.precedence}")
+
+        res.append("\n    ")
+        res.append_text(repr_pyfunction(self.implementation))
+
+        return res
+
+
+@rich_repr
+class MethodList(list):
+    "A list of :class:`Method`s which is nicely printed by :mod:`rich`."
+
+    def __rich_console__(self, console, options):
+        yield f"List of {len(self)} method(s):"
+        for i, method in enumerate(self):
+            yield Padding(Text(f"[{i}] ") + method.repr_mismatch(), (0, 4))
```

Comment:
Sorry, that was confused.

I think here it should yield the method itself (which will call `methd.__rich_console__`) instead of `repr_mismatch`.

Reply:
`repr_mismatch` currently gives back a list of `rich` objects. I should look more closely at `rich` how precisely all of this works...

I tried something like

```
            yield Padding(Text(f"[{i}] "), (0, 4))
            yield Padding(method)
```

but that puts the two on different lines.

Just adding them seems to do the right thing

Reply:
I suspect you could safely do
```python
Padding(Text(f"[{i}] ") + method. __rich_console__(console, options):(), (0, 4))
```
because rich console of methods are text objects which can be concatenated.
Alternatively
```
Padding(Text(f"[{i}] ").append(method. __rich_console__(console, options)):(), (0, 4))
```

Reply:
That worked! Though I had to sum the generated elements:

```python
            method_repr = method.__rich_console__(console, options)
            yield Padding(sum(method_repr, Text(f"[{i}] ")), (0, 4))
```